runtime/lua/vim/lsp/client.lua

```diff
@@ -509,6 +509,13 @@ function Client:initialize()
     root_path = vim.uri_to_fname(root_uri)
   end
 
+  -- Ensure empty tables for initializationOptions are serialized as objects, not arrays
+  local init_options = config.init_options
+  if type(init_options) == 'table' and vim.tbl_isempty(init_options) then
+    -- For empty tables, ensure they're serialized as objects
+    init_options = vim.empty_dict()
```

Comment:
As Mathias mentioned, a server may require an array, so we can't assume dict here.

Thanks for the thoughtful PR, though.